### PR TITLE
Location management and process chain validation

### PR DESCRIPTION
**Pull request by anikaweinmann**
Added:
- location managment:
  - delete
  - create
- process chain validation
- restructuring of the docs
- restructuring of the tests
- deploy workflow for docs

### CODE REVIEW

File: actinia/actinia.py [hunk 1]
```diff
@@ -164,16 +165,35 @@ def __request_locations(self):
...
-    # def create_location(self, name, epsgcode):
-    #     """
-    #     Creates a new location with given name and epsgcode via post request.
-    #
-    #
-    #     :return: The created location
-    #     """
+    def create_location(self, name, epsgcode):
+        """
+        Creates a new location with given name and epsgcode via post request.
```
Comment (neteler) [inline actinia/actinia.py:170]:
```suggestion
        Creates a new location with given name and EPSG code via post request.
```

File: actinia/location.py [hunk 2]
```diff
@@ -104,6 +114,55 @@ def get_mapsets(self):
...
+        elif resp.status_code == 400:
+            msg = f"Validation error: {json.loads(resp.text)['message']}"
+            print(msg, file=sys.stderr)
+        else:
+            raise Exception(f"Error {resp.status_code}: {resp.text}")
+
+    def validate_process_chain_async(self, pc, name=None):
+        """Validate a process chain (async)."""
+        actiniaResp = self.__validate_process_chain(pc, "async")
+        orig_name, name = self.__set_job_names(name, "unkonwn_validation_job")
```
Comment (neteler) [inline actinia/location.py:157]:
```suggestion
        orig_name, name = self.__set_job_names(name, "unknown_validation_job")
```

File: docs/docs/05_mapset_managment.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Mapsets Management
+
+With the mapset management the mapsets of a specified location can be
+requested as well as information of each mapset.
+Also a mapsest can be created and deleted if the user is permitted.
+
+First connecting actinia Python library with [actinia](https://actinia.mundialis.de/) and set authentication (Attention: The demouser is not permitted to create or delete a mapset!)
```
Comment (neteler) [inline docs/docs/05_mapset_managment.md:7]:
```suggestion
First connecting actinia Python library with [actinia](https://actinia.mundialis.de/) and
set authentication (Attention: The demouser is not permitted to create or delete a mapset!)
```

File: docs/docs/05_mapset_managment.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+## Get Mapsets of Specified Location
+Get mapsets of the ***nc_spm_08*** location:
+```
+mapsets = actinia_mundialis.locations["nc_spm_08"].get_mapsets()
+print(mapsets.keys())
+```
+
+## TODO: Create Mapset in Specified Location
+
+## TODO: Delete Mapset in Specified Location
```
Comment (neteler) [inline docs/docs/05_mapset_managment.md:28]:
```suggestion
## Get mapsets of specified location
Get mapsets of the ***nc_spm_08*** location:
```
mapsets = actinia_mundialis.locations["nc_spm_08"].get_mapsets()
print(mapsets.keys())
```

## TODO: Create new mapset in specified location

## TODO: Delete mapset in specified location
```

File: docs/docs/06_process_chain_validation.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          ]
+      }
+    ],
+    "version": "1"
+}
+pc = {"list": [{"id": "r_mapcalc","module": "r.mapcalc","inputs": [{"param": "expression","value": "baum=5"}]}],"version": "1"}
+actinia_mundialis.locations["nc_spm_08"].validate_process_chain_sync(pc)
+```
+
+## A-synchron process chain valitaion:
```
Comment (neteler) [inline docs/docs/06_process_chain_validation.md:38]:
```suggestion
## Asynchronous process chain validation:
```

File: docs/docs/06_process_chain_validation.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+```
+pc = {
+    "list": [
+      {
+          "id": "r_mapcalc",
+          "module": "r.mapcalc",
+          "inputs": [
+              {
+                  "param": "expression",
+                  "value": "baum=5"
```
Comment (neteler) [inline docs/docs/06_process_chain_validation.md:27]:
```suggestion
                  "value": "elevation=42"
```

File: docs/docs/06_process_chain_validation.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+              }
+          ]
+      }
+    ],
+    "version": "1"
+}
+pc = {"list": [{"id": "r_mapcalc","module": "r.mapcalc","inputs": [{"param": "expression","value": "baum=5"}]}],"version": "1"}
```
Comment (neteler) [inline docs/docs/06_process_chain_validation.md:34]:
```suggestion
pc = {"list": [{"id": "r_mapcalc","module": "r.mapcalc","inputs": [{"param": "expression","value": "elevation=42"}]}],"version": "1"}
```

File: tests/test_actinia_mapset_management.py [hunk 8]
```diff
@@ -31,53 +31,67 @@
...
-__license__ = "TODO"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
 
+LOCATION_NAME = "nc_spm_08"
+PC = {
+    "list": [
+        {
+            "id": "r_mapcalc",
+            "module": "r.mapcalc",
+            "inputs": [{"param": "expression", "value": "baum=5"}],
```
Comment (neteler) [inline tests/test_actinia_mapset_management.py:53]:
```suggestion
            "inputs": [{"param": "expression", "value": "elevation=42"}],
```

File: tests/test_process_chain_validation.py [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2022, mundialis GmbH & Co. KG"
+
+LOCATION_NAME = "nc_spm_08"
+PC = {
+    "list": [
+        {
+            "id": "r_mapcalc",
+            "module": "r.mapcalc",
+            "inputs": [{"param": "expression", "value": "baum=5"}],
```
Comment (neteler) [inline tests/test_process_chain_validation.py:60]:
```suggestion
            "inputs": [{"param": "expression", "value": "elevation=42"}],
```

File: docs/docs/07_processing.md [hunk 10]
```diff
@@ -1,38 +1,17 @@
-# Quickstart
+# Processing
 
-Connecting actinia Python library with [actinia](https://actinia.mundialis.de/)
+Start a process chain with a valid process chain.
```
Comment (neteler) [inline docs/docs/07_processing.md:3]:
```suggestion
Start a process chain with a valid process chain.
```

This sentence is not clear to me...

File: tests/test_actinia_location_management.py [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    def mock_delete_location(cls):
+        cls.mock_delete.return_value = Mock()
+        cls.mock_delete.return_value.status_code = 200
+        cls.mock_delete.return_value.text = json.dumps(delete_location_resp)
+
+    def test_actinia_get_locations(self):
+        """Test get locations."""
+
+        assert self.testactinia.locations == {}, (
+            "Locations are not empty dictionary"
```
Comment (neteler) [inline tests/test_actinia_location_management.py:95]:
```suggestion
            "Locations may not be an empty dictionary"
```

(not sure, a suggestion)

File: docs/docs/07_processing.md (hunk 10 above)
Comment (anikaweinmann) [reply docs/docs/07_processing.md]: I changed it

**Pull request merged**